stdlib/Dates/src/rounding.jl

```diff
@@ -85,22 +85,21 @@ function Base.floor(dt::DateTime, p::TimePeriod)
 end
 
 """
-    floor(x::Period, precision::T) where T <: Union{TimePeriod, Week, Day} -> T
+    floor(precision::T, x::Period) where T <: Union{TimePeriod, Week, Day} -> T
 
 Round `x` down to the nearest multiple of `precision`. If `x` and `precision` are different
 subtypes of `Period`, the return value will have the same type as `precision`.
 
-For convenience, `precision` may be a type instead of a value: `floor(x, Dates.Hour)` is a
-shortcut for `floor(x, Dates.Hour(1))`.
+Warning: Argument order of this floor method differs from the standard rounding argument order.
```

Comment:
Not quite. The argument order is normal for `floor(Minute, Second(63))` but reversed for `floor(Second(63), Second(15))`. That subtle distinction is precisely what we need to make users aware of (and fix if Julia 2.0 ever comes around).

Also, use this syntax for warnings ([example](https://github.com/JuliaLang/julia/blob/7a21d52d4847a3ebed944888eae98b8c69472861/base/experimental.jl#L6)) so that automated tools (e.g. the REPL and web hosted documentation) provide special rendering:
```
!!! warning
    foo is not the same as bar
```

Reply:
@LilithHafner Added !!! warning. Its also returning proper values this time. Can you review the code?

Reply:
Your changes broke the doc build and unfortunately, since you force pushed, I can't see the diff to point to the specific reason why.

The warning also still does correctly describe the argument order as will be.

Reply:
@LilithHafner May I know, what should `floor(Minute(15), Minute(44) `and `floor(Minute, Minute(44))` return?

Reply:
In 1.8 we have this behavior
```julia
julia> VERSION
v"1.8.3"

julia> floor(Minute(15), Minute(44))
0 minutes

julia> floor(Minute, Minute(44))
ERROR: MethodError: no method matching floor(::Type{Minute}, ::Minute)
Closest candidates are:
  floor(::Union{Day, Week, TimePeriod}, ::T) where T<:Union{Day, Week, TimePeriod} at ~/.julia/juliaup/julia-1.8.3+0.x64.apple.darwin14/share/julia/stdlib/v1.8/Dates/src/rounding.jl:110
  floor(::Type{T}, ::Missing) where T at missing.jl:154
  floor(::Type{T}, ::Rational) where T at rational.jl:458
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[6]:1
```
That answers your first question: **`floor(Minute(15), Minute(44))` must return `Minute(0)`** for backwards compatibility.

For your second question, **`floor(Minute, Minute(44))`** errored in 1.8, so we can give it new behavior in 1.10. I think it **should return `Minute(44)`**.

Reply:
@LilithHafner Can you check if the warning is more precise now?

Reply:
The example of `floor(Minute(15),Minute(44))` shows that this deprecation might not be a good idea after all. Of course I'd prefer consistency of rounding functions, but introducing contradicting behaviour therefore isn't worth it, given that until now this was documented:
`For convenience, p may be a type instead of a value: floor(dt, Dates.Hour) is a shortcut for floor(dt, Dates.Hour(1)).`